src/System/Wlog/Logger.hs

```diff
@@ -124,30 +123,30 @@ Example return value:
 >["", "MissingH", "System.Cmd.Utils", "System.Cmd.Utils.pOpen"]
 
 -}
-componentsOfName :: String -> [String]
+componentsOfName :: Text -> [Text]
 componentsOfName name =
-    rootLoggerName : joinComp (split "." name) []
+    rootLoggerName : joinComp (T.splitOn "." name) ""
   where
     joinComp [] _ = []
-    joinComp (x:xs) [] = x : joinComp xs x
+    joinComp (x:xs) "" = x : joinComp xs x
     joinComp (x:xs) accum =
-        let newlevel = accum ++ "." ++ x
+        let newlevel = accum <> "." <> x
         in newlevel : joinComp xs newlevel
 
 ---------------------------------------------------------------------------
 -- Logging With Location
 ---------------------------------------------------------------------------
 
 -- | Log a message using the given logger at a given priority.
-logM :: String     -- ^ Name of the logger to use
+logM :: Text       -- ^ Name of the logger to use
```

Comment:
And here. Basically, it's really good idea to use `LoggerName` newtype everywhere you intend to use `LoggerName`. So it won't be so easy to mess up with function arguments.